gen/utils/data/data.go

```diff
@@ -0,0 +1,181 @@
+package data
+
+import (
+	"bytes"
+	"crypto/tls"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"slices"
+	"strconv"
+	"strings"
+
+	"github.com/CiscoDevNet/terraform-provider-aci/v2/gen/utils"
+	"github.com/CiscoDevNet/terraform-provider-aci/v2/gen/utils/logger"
+)
+
+// Initialize a logger instance for the generator.
+var genLogger = logger.InitializeLogger()
+
+type DataStore struct {
+	// A map containing all the information about the classes required to render the templates.
+	Classes map[string]Class
```

Comment:
Does it need to be public?
```suggestion
	classes map[string]Class
```

Reply:
for now made it public because outside the data the classes should be accessed, can change and create method to loop through and retrieve information from the map if you think that would be better.